darts/models/forecasting/transformer_model.py

```diff
@@ -276,47 +278,118 @@ def __init__(
             custom_decoder=custom_decoder,
         )
 
-        self.decoder = nn.Linear(
-            d_model, self.target_length * self.target_size * self.nr_params
-        )
+        self.decoder = nn.Linear(d_model, self.target_size * self.nr_params)
 
-    def _create_transformer_inputs(self, data):
+    def _permute_transformer_inputs(self, data):
```

Comment:
if this method is reduced to a simple permutation of the dimensions, it would be better to get ride of it (and keep the comment)